crates/storage_impl/src/lib.rs

```diff
@@ -222,6 +283,79 @@ impl DataModelExt for data_models::MerchantStorageScheme {
     }
 }
 
+impl RedisConnInterface for MockDb {
+    fn get_redis_conn(
+        &self,
+    ) -> Result<Arc<redis_interface::RedisConnectionPool>, error_stack::Report<RedisError>> {
+        Ok(self.redis.clone())
+    }
+}
+
+#[cfg(feature = "kms")]
+/// Store the decrypted kms secret values for active use in the application
+/// Currently using `StrongSecret` won't have any effect as this struct have smart pointers to heap
+/// allocations.
+/// note: we can consider adding such behaviour in the future with custom implementation
+#[derive(Clone)]
+pub struct ActiveKmsSecrets {
+    pub jwekey: masking::Secret<Jwekey>,
+}
+#[derive(Debug, Deserialize, Clone, Default)]
+#[serde(default)]
+pub struct Jwekey {
```

Comment:
does this need to be part of storage_impl ?

we can move it to common_utils if it's also used in scheduler else we can retain it in router